### PR TITLE
Power law creep in tensor_mechanics.

### PR DESCRIPTION
**Pull request by jasondhales**
Implement power law creep in tensor_mechanics.

### CODE REVIEW

File: modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C [hunk 1]
```diff
@@ -116,3 +118,12 @@ RecomputeRadialReturn::computeInelasticStrainIncrement()
 
   computeStressFinalize(_inelastic_strain_increment[_qp]);
 }
+
+Real
+RecomputeRadialReturn::getIsotropicShearModulus()
+{
+  Real shear_modulus = _elasticity_tensor[_qp](1,2,1,2);
+  if ((_mesh.dimension() == 3) && (shear_modulus != _elasticity_tensor[_qp](1,3,1,3)))
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C]: the inner bracket pairs are not needed

**Comment by dschwen** [comment]
Looks good to me.

File: modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C (hunk 1 above)
Comment (jasondhales) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C]: True.  This was code moved from RecomputeRadialReturnIsotropicPlasticity.  Want me to remove them?

Comment (sapitts) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturn.C]: Please do--thank you!

File: modules/tensor_mechanics/src/materials/RecomputeRadialReturnPowerLawCreep.C [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+void
+RecomputeRadialReturnPowerLawCreep::computeStressInitialize(Real effectiveTrialStress)
+{
+  _shear_modulus = getIsotropicShearModulus();
+
+  _exponential = 1;
+  if (_has_temp)
+  {
+    _exponential = std::exp(-_activation_energy / (_gas_constant * _temperature[_qp]));
+  }
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/RecomputeRadialReturnPowerLawCreep.C]: If you push changes anyways... Single statement if bodies should not have curly braces. And I think an `if` `else` would be a bit more expressive.

```
+  _exponential = 1;
  if (_has_temp)
    _exponential = std::exp(-_activation_energy / (_gas_constant * _temperature[_qp]));
  else
    _exponential = 1.0;
```

File: modules/tensor_mechanics/include/materials/RecomputeRadialReturnPowerLawCreep.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * in conjunction with ComputeRadialReturnMappingStress.  This class calculates
+ * an effective trial stress, an effective scalar plastic strain
+ * increment, and the derivative of the scalar effective plastic strain increment;
+ * these values are passed to the RecomputeRadialReturnStressIncrement to compute
+ * the radial return stress increment.  This isotropic plasticity class also
+ * computes the plastic strain as a stateful material property.
+ *
+ * This class is based on the implicit integration algorithm in F. Dunne and N.
+ * Petrinic's Introduction to Computational Plasticity (2004) Oxford University
+ * Press, pg. 146 - 149.
```
Comment (sapitts) [inline modules/tensor_mechanics/include/materials/RecomputeRadialReturnPowerLawCreep.h]: Please update the comments to reflect the creep model from the isotropic plasticity model.  The generic viscoplasticity algorithm is given on pg 161-163 of the Dunne and Petrinic book.  Thank you!

**Comment by jasondhales** [comment]
The debug linux gnu recipe failed due to a timeout in a test not related to this PR.  What now?

**Comment by dschwen** [comment]
Somebody already invalidated the test.

**Comment by permcody** [comment]
👍

**Pull request merged**